apps/allocations/app/components/Panel/NewAllocation.js

```diff
@@ -66,23 +59,48 @@ class NewAllocation extends React.Component {
     this.setState({ payoutTypeIndex: index, payoutType: items[index] })
   }
 
+  // Should be using web3.isAddress probably but this is good enough for now
+  isAddress = addr => {
+    if (!/^(0x)?[0-9a-f]{40}$/i.test(addr)) {
+      return false
+    }
+    return true
+  }
+
   submitAllocation = () => {
     // clear input here.
+
     let informational = (this.state.allocationTypeIndex === 0)
     let recurring = !informational && this.state.payoutTypeIndex != 0
     // TODO: period should be smarter: now the only option is monthly
     let period = recurring ? 86400 * 31 : 0
-
+    let optionsInput = this.state.optionsInput
+    this.setState({
+      addressError: this.state.addressError,
+      allocationError: false,
+    })
+    if (!(this.isAddress(optionsInput) || optionsInput === '')) {
+      this.setState({ addressError: true })
+      this.setState()
+      return
+    }
+    if (this.isAddress(optionsInput)) {
+      this.state.options.push(this.state.optionsInput)
```

Comment:
What you probably want here is to use the [function argument form of `setState()`](https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous) instead (and as a rule, **never** directly modify state).

Reply:
Fixed